upgrades/upgrade_testing_framework/steps/step_test_source_cluster.py

```diff
@@ -0,0 +1,21 @@
+from upgrade_testing_framework.core.framework_step import FrameworkStep
+import upgrade_testing_framework.core.shell_interactions as shell
+from upgrade_testing_framework.cluster_management.node import Node
+
+
+class TestSourceCluster(FrameworkStep):
+    """
+    This step is where you run tests on the source cluster
+    """
+
+    def _run(self):
+        # Get the state we need
+        # N/A
+
+        # Begin the step body
+        _, output = shell.call_shell_command("curl -X GET \"localhost:9200/_cat/nodes?v=true&pretty\"")
```

Comment:
a just curious question: any particular reason to do this via shell command vs. opensearch-py library? (thought is that the python client has a fair amount of error handling/possibly retries built in already)

(actual change would be out of scope of this PR, but for the future)

Reply:
No particular reason than this took me ~1 min to write rather than ~30 minutes.  Pulling the OpenSearch SDK into the Framework is something we should do eventually.